pipeline.yaml

```diff
@@ -30,60 +35,73 @@ data:
-data_scoring:
-  location: {{INGEST_SCORING_DATA_LOCATION}}
-  format: {{INGEST_SCORING_DATA_FORMAT|default('parquet')}}
-  custom_loader_method: steps.ingest.load_file_as_dataframe
-  sql: SELECT * FROM delta.`{{INGEST_SCORING_DATA_LOCATION}}`
-# Specifies the name of the column containing targets / labels for model training and evaluation
-target_col: "fare_amount"
+#
+# FIXME::REQUIRED: Specifies the target column name for model training and evaluation.
+#
+target_col: ""
 steps:
   split:
-    # Train/validation/test split ratios
-    split_ratios: {{SPLIT_RATIOS|default([0.75, 0.125, 0.125])}}
-    # Specifies the method to use to perform additional cleaning on split datasets
-    # Note that arbitrary transformations should go into the transform step
+    #
+    # FIXME::OPTIONAL: Adjust the train/validation/test split ratios below.
+    #
+    split_ratios: [0.75, 0.125, 0.125]
+    #
+    #  FIXME::OPTIONAL: Specifies the method to use to "post-process" the split datasets. Note that
+    #                   arbitrary transformations should go into the transform step.
     post_split_method: steps.split.process_splits
   transform:
-    # Specifies the method that defines the data transformations to apply during model inference
+    #
+    #  FIXME::OPTIONAL: Specifies the method that defines an sklearn-compatible transformer, which
+    #                   applies input feature transformation during model training and inference.
     transformer_method: steps.transform.transformer_fn
   train:
     using: estimator_spec
     # Specifies the method that defines the estimator type and parameters to use for model training
     estimator_method: steps.train.estimator_fn
   evaluate:
-    # Sets performance thresholds that a trained model must meet in order to be eligible for
-    # registration to the MLflow Model Registry
-    validation_criteria:
-      - metric: root_mean_squared_error
-        threshold: 10
-      - metric: mean_absolute_error
-        threshold: 50
-      - metric: weighted_mean_squared_error
-        threshold: 20
+    #
+    # FIXME::OPTIONAL: Sets performance thresholds that a trained model must meet in order to be
+    #                  eligible for registration to the MLflow Model Registry.
+    #
+    # validation_criteria:
+    #   - metric: root_mean_squared_error
+    #     threshold: 10
   register:
-    # Specifies the name of the Registered Model to use when registering a trained model to
-    # the MLflow Model Registry
-    model_name: "taxi_fare_regressor"
+    #
+    # FIXME::REQUIRED: Specifies the name of the Registered Model to use when registering a trained
+    #                  model to the MLflow Model Registry.
+    #
+    model_name: ""
     # Indicates whether or not a model that fails to meet performance thresholds should still
     # be registered to the MLflow Model Registry
     allow_non_validated_model: false
   predict:
```

Comment:
Is there a reason we need this if everything below is optional, can we comment this out? Do we need to fix mlflow code for this?

Reply:
MLflow needs to be fixed with @prithvikannan 's PR. The template doesn't work as is. I'm assuming it'll be fixed soon.
We can comment `predict` step out but I want to leave it there to indicate such step actually exists.